test/plugin.test.js

```diff
@@ -94,6 +94,32 @@ QUnit.test('play() without a src is an error', function(assert) {
     'error type is no source');
 });
 
+QUnit.test('no progress for 1 second shows the loading spinner', function(assert) {
+  this.player.src(sources);
+  this.player.trigger('play');
+  this.clock.tick(1 * 1000);
+
+  assert.ok(
+    this.player.hasClass('vjs-waiting'),
+    'the plugin adds spinner class to the player'
+  );
+});
+
+QUnit.test('progress events while playing reset the spinner', function(assert) {
+  this.player.src(sources);
+  this.player.trigger('play');
+  // stalled for awhile
+  this.clock.tick(44 * 1000);
+  assert.ok(
+    this.player.hasClass('vjs-waiting'),
+    'the plugin adds spinner class to the player'
+  );
+
+  // but playback resumes!
+  this.player.trigger('progress');
```

Comment:
I don't think this is what `progress` indicates. From the spec:

> The user agent is fetching media data.

An alternative may be to track whether the media is actually playing back by checking that the current time changes by a minimum threshold.